### PR TITLE
Add priorityClassName to node pods

### PR DESCRIPTION
**Pull request by Bhagyashreek8**
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [x] I have made sure existing UTs are not impacted
- [ ] I have executed all necessary linter and made sure internal pipeline/travis is not broken

### CODE REVIEW

**Comment by Bhagyashreek8** [comment]
csi-node pod describe - you can observe priorityClassName set as below
```
Priority:             2000001000
Priority Class Name:  system-node-critical
```


```
% kubectl describe pod ibm-object-csi-node-nddmd -n ibm-object-csi-operator
Name:                 ibm-object-csi-node-nddmd
Namespace:            ibm-object-csi-operator
Priority:             2000001000
Priority Class Name:  system-node-critical
Service Account:      ibm-object-csi-node-sa
Node:                 10.240.0.12/10.240.0.12
Start Time:           Wed, 11 Sep 2024 21:58:54 +0530
Labels:               app.kubernetes.io/component=node
                      app.kubernetes.io/instance=ibm-object-csi
                      app.kubernetes.io/managed-by=ibm-object-csi-driver-operator
                      app.kubernetes.io/name=ibm-object-csi-driver
                      app.kubernetes.io/part-of=ibm-object-csi-driver
                      app.kubernetes.io/version=1.0.0
                      controller-revision-hash=6c4845b45f
                      pod-template-generation=1
                      product=ibm-object-csi-driver
                      release=v1.0.0
Annotations:          cni.projectcalico.org/containerID: 3280aa071d338156db6709f6cb24993e7e30b37f4e9d2a2091ab2ef86d746905
                      cni.projectcalico.org/podIP: 172.17.118.9/32
                      cni.projectcalico.org/podIPs: 172.17.118.9/32
                      productID: ibm-object-csi-driver
                      productName: ibm-object-csi-driver
                      productVersion: 1.0.0
Status:               Running
IP:                   172.17.118.9
IPs:
  IP:           172.17.118.9
Controlled By:  DaemonSet/ibm-object-csi-node
Containers:
  ibm-object-csi-node:
    Container ID:  containerd://dcbc6f804e5bb484fd2de210513f66d4b4d110619d29315d0347f5ad812643a0
    Image:         icr.io/ibm/ibm-object-csi-driver:v0.1.8
    Image ID:      icr.io/ibm/ibm-object-csi-driver@sha256:b7ba40ff6c0cf1052f765496bad0ecc26111a322479832c6a888ad68e3fa1d94
    Port:          9808/TCP
    Host Port:     0/TCP
    Args:
      --servermode=node
      --endpoint=$(CSI_ENDPOINT)
      --nodeid=$(KUBE_NODE_NAME)
      --logtostderr=true
      --v=5
    State:          Running
      Started:      Wed, 11 Sep 2024 21:59:03 +0530
    Ready:          True
    Restart Count:  0
    Liveness:       http-get http://:9808/healthz delay=10s timeout=3s period=10s #success=1 #failure=30
    Environment:
      CSI_ENDPOINT:    unix:///csi/csi.sock
      KUBE_NODE_NAME:   (v1:spec.nodeName)
    Mounts:
      /csi from plugin-dir (rw)
      /dev/fuse from fuse-device (rw)
      /dev/log from log-dev (rw)
      /host/var/log from host-log (rw)
      /var/data/kubelet from kubelet-dir-ibm (rw)
      /var/lib/kubelet from kubelet-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-pn5hs (ro)
  csi-node-driver-registrar:
    Container ID:  containerd://08d73ad840beb120c6d5466449fc7261d3bbf7a6239e48e65f269e2dfe35b65d
    Image:         k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.6.3
    Image ID:      k8s.gcr.io/sig-storage/csi-node-driver-registrar@sha256:fdff3ee285341bc58033b6b2458a5d45fd90ec6922a8ba6ebdd49b0c41e2cd34
    Port:          <none>
    Host Port:     <none>
    Args:
      --csi-address=$(ADDRESS)
      --kubelet-registration-path=$(DRIVER_REG_SOCK_PATH)
      --v=5
    State:          Running
      Started:      Wed, 11 Sep 2024 21:59:06 +0530
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     40m
      memory:  80Mi
    Requests:
      cpu:     10m
      memory:  20Mi
    Environment:
      ADDRESS:               /csi/csi.sock
      DRIVER_REG_SOCK_PATH:  /var/lib/kubelet/plugins/cos.s3.csi.ibm.io/csi.sock
    Mounts:
      /csi from plugin-dir (rw)
      /registration from registration-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-pn5hs (ro)
  livenessprobe:
    Container ID:  containerd://6a0703565684772acec85964bc5a1b9622e7466c395e85cef1babed87706b823
    Image:         k8s.gcr.io/sig-storage/livenessprobe:v2.9.0
    Image ID:      k8s.gcr.io/sig-storage/livenessprobe@sha256:2b10b24dafdc3ba94a03fc94d9df9941ca9d6a9207b927f5dfd21d59fbe05ba0
    Port:          <none>
    Host Port:     <none>
    Args:
      --csi-address=/csi/csi.sock
      --health-port=9808
    State:          Running
      Started:      Wed, 11 Sep 2024 21:59:08 +0530
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     20m
      memory:  40Mi
    Requests:
      cpu:        10m
      memory:     10Mi
    Environment:  <none>
    Mounts:
      /csi from plugin-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-pn5hs (ro)
Conditions:
  Type                        Status
  PodReadyToStartContainers   True 
  Initialized                 True 
  Ready                       True 
  ContainersReady             True 
  PodScheduled                True 
Volumes:
  kubelet-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/kubelet
    HostPathType:  Directory
  plugin-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/kubelet/plugins/cos.s3.csi.ibm.io
    HostPathType:  DirectoryOrCreate
  registration-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/kubelet/plugins_registry
    HostPathType:  Directory
  kubelet-dir-ibm:
    Type:          HostPath (bare host directory volume)
    Path:          /var/data/kubelet
    HostPathType:  DirectoryOrCreate
  fuse-device:
    Type:          HostPath (bare host directory volume)
    Path:          /dev/fuse
    HostPathType:  
  log-dev:
    Type:          HostPath (bare host directory volume)
    Path:          /dev/log
    HostPathType:  
  host-log:
    Type:          HostPath (bare host directory volume)
    Path:          /var/log
    HostPathType:  
  kube-api-access-pn5hs:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/disk-pressure:NoSchedule op=Exists
                             node.kubernetes.io/memory-pressure:NoSchedule op=Exists
                             node.kubernetes.io/not-ready:NoExecute op=Exists
                             node.kubernetes.io/pid-pressure:NoSchedule op=Exists
                             node.kubernetes.io/unreachable:NoExecute op=Exists
                             node.kubernetes.io/unschedulable:NoSchedule op=Exists
Events:
  Type    Reason     Age    From               Message
  ----    ------     ----   ----               -------
  Normal  Scheduled  4m18s  default-scheduler  Successfully assigned ibm-object-csi-operator/ibm-object-csi-node-nddmd to 10.240.0.12
  Normal  Pulling    4m17s  kubelet            Pulling image "icr.io/ibm/ibm-object-csi-driver:v0.1.8"
  Normal  Pulled     4m9s   kubelet            Successfully pulled image "icr.io/ibm/ibm-object-csi-driver:v0.1.8" in 8.089s (8.089s including waiting). Image size: 124805285 bytes.
  Normal  Created    4m9s   kubelet            Created container ibm-object-csi-node
  Normal  Started    4m9s   kubelet            Started container ibm-object-csi-node
  Normal  Pulling    4m9s   kubelet            Pulling image "k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.6.3"
  Normal  Pulled     4m6s   kubelet            Successfully pulled image "k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.6.3" in 2.656s (2.656s including waiting). Image size: 10124505 bytes.
  Normal  Created    4m6s   kubelet            Created container csi-node-driver-registrar
  Normal  Started    4m6s   kubelet            Started container csi-node-driver-registrar
  Normal  Pulling    4m6s   kubelet            Pulling image "k8s.gcr.io/sig-storage/livenessprobe:v2.9.0"
  Normal  Pulled     4m4s   kubelet            Successfully pulled image "k8s.gcr.io/sig-storage/livenessprobe:v2.9.0" in 2.108s (2.108s including waiting). Image size: 9194114 bytes.
  Normal  Created    4m4s   kubelet            Created container livenessprobe
  Normal  Started    4m4s   kubelet            Started container livenessprobe
bhagyashreek@Bhagyashrees-MacBook-Pro ibm-object-csi-driver-operator % 
bhagyashreek@Bhagyashrees-MacBook-Pro ibm-object-csi-driver-operator % 
```

**Comment by Bhagyashreek8** [comment]
Verify pvc mount
```
% kubectl create -f s3fs-secret.yaml
secret/s3fs-pvc created
bhagyashreek@Bhagyashrees-MacBook-Pro cos-csi-addon % 
bhagyashreek@Bhagyashrees-MacBook-Pro cos-csi-addon % 
bhagyashreek@Bhagyashrees-MacBook-Pro cos-csi-addon % kubectl get secrets
NAME                                           TYPE                DATA   AGE
bha-p2-3ccc0517303a146ee7d816741933e8cf-0000   kubernetes.io/tls   2      116m
s3fs-pvc                                       cos-s3-csi-driver   5      8s
bhagyashreek@Bhagyashrees-MacBook-Pro cos-csi-addon % 
bhagyashreek@Bhagyashrees-MacBook-Pro cos-csi-addon % 
bhagyashreek@Bhagyashrees-MacBook-Pro cos-csi-addon % kubectl create -f s3fs-pvc.yaml
persistentvolumeclaim/s3fs-pvc created
bhagyashreek@Bhagyashrees-MacBook-Pro cos-csi-addon % 
bhagyashreek@Bhagyashrees-MacBook-Pro cos-csi-addon % 
bhagyashreek@Bhagyashrees-MacBook-Pro cos-csi-addon % kubectl get pvc
NAME       STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                       VOLUMEATTRIBUTESCLASS   AGE
s3fs-pvc   Bound    pvc-812c9b4c-a9b6-4a53-9a5e-45532cb8be73   256Mi      RWO            ibm-object-storage-standard-s3fs   <unset>                 12s
bhagyashreek@Bhagyashrees-MacBook-Pro cos-csi-addon % 


% kubectl get pods                                      
NAME                                READY   STATUS    RESTARTS   AGE
cos-csi-test-app-656d6c5575-6pfqg   1/1     Running   0          58s
bhagyashreek@Bhagyashrees-MacBook-Pro cos-csi-addon % 
bhagyashreek@Bhagyashrees-MacBook-Pro cos-csi-addon % 
bhagyashreek@Bhagyashrees-MacBook-Pro cos-csi-addon % kubectl exec -it cos-csi-test-app-656d6c5575-6pfqg -- /bin/bash
root@cos-csi-test-app-656d6c5575-6pfqg:/# mount | grep s3fs
s3fs on /mnt/s3fsvolume type fuse.s3fs (rw,nosuid,nodev,relatime,user_id=0,group_id=0,allow_other)
root@cos-csi-test-app-656d6c5575-6pfqg:/# cd mnt/s3fsvolume/
root@cos-csi-test-app-656d6c5575-6pfqg:/mnt/s3fsvolume# ls
test.txt  test2.txt
root@cos-csi-test-app-656d6c5575-6pfqg:/mnt/s3fsvolume# cat test.txt 
Addon v0.1.10 verification
root@cos-csi-test-app-656d6c5575-6pfqg:/mnt/s3fsvolume# cat test2.txt 
Addon v0.1.11
root@cos-csi-test-app-656d6c5575-6pfqg:/mnt/s3fsvolume# echo "priorityClassName test" > test.txt 
root@cos-csi-test-app-656d6c5575-6pfqg:/mnt/s3fsvolume# cat test.txt 
priorityClassName test
root@cos-csi-test-app-656d6c5575-6pfqg:/mnt/s3fsvolume# exit
exit
```

**Review by mssachan — DISMISSED** [review]
lgtm

File: README.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
 # ibm-object-csi-driver-operator
-// TODO(user): Add simple overview of use/purpose
+A CSI based object storage plugin with dynamic bucket provisioning and plugable mounters, like rclone, goofys, s3fs and other. The plugin should seamlessly work in IBM, AWS and GCP platforms.
```
Comment (nkkashyap) [inline README.md:2]: Pls remove `goofyfs`

**Review by nkkashyap — APPROVED** [review]
lgtm

**Review by ashimagarg27 — APPROVED** [review]

**Pull request merged**